### ISSUE TITLE
cross vlan monitoring

### ISSUE THREAD

**Issue by topeminent** [veyon/veyon#325]
Hi, 

I m  trying to configure veyon across multiple vlans, as my classrooms are on diff. vlans. I can ping, RD, wol, these but veyon 4.0.7 can t connect to these pc s.

Did anyone managed to do something like this?

Any input would be appreciated.
Steve.

**Comment by tobydox** [comment]
Does it work within the same VLAN? If so, did you perform the tests as described in the troubleshooting section of the admin manual?

**Comment by ldlx** [comment]
Is there a router in between? Are the computers on the same subnet or different ones?

**Comment by topeminent** [comment]
@tobydox Yes, it works on the same vlan. I use key file authentication. Also i did the test described in troubleshooting admin manual

**Comment by topeminent** [comment]
> Is there a router in between? Are the computers on the same subnet or different ones?

Yes there is, the computers are on diffrent subnets. I have proper forwarding rules set up, thats why i can ping, RD, and wol from management vlan to my classroom vlan

**Comment by ldlx** [comment]
Do you use AD to populate the rooms or did you create rooms/computers manually? Is name resolution working properly (Hostname and/or DNS name)? Can you connect to computers via IP address?

Btw. how do you manage WoL in different subnets? Do you forward broadcast packages to all subnets?

**Comment by topeminent** [comment]
I populate them manually, i m currently in test faze if it s a viable road for our infrastructure. I connect via ip addresses, no dns at all. I can manage wol on diffrent subnets, via router configuration, using arp proxy option.

**Comment by topeminent** [comment]
> can you connect when you are on the same vlan/subnet as the client computer?

Yes i can.

**Comment by ldlx** [comment]
Is there an active software firewall on the clients? On Windows you may need to enable edge traversal ("Allow edge traversal") on the advanced tab of the firewall rule.

**Comment by topeminent** [comment]
there is, an eset endpoint security firewall, which i tuned it off. Could somebody confirm, that it should work cross vlan , or did somebody successfully test this ? It s hard to believe, that no one segments their internal network using vlans.

**Comment by ldlx** [comment]
Confirmed :-) I have such environment up and running, switched from iTalc to Veyon two weeks ago.

**Comment by topeminent** [comment]
Thx for the feedback. I ll follow up when i have new information.

**Comment by adgm1** [comment]
I can you be more precise to configure the settings of the firewall to be abble to wol from different vlan with VEYON? thanks

**Comment by topeminent** [comment]
WOL uses UDP port 9, so you should make sure that port is opened on your router. I use a mikrotik router, so i had to configure arp-proxy on it, so the router will forward the WOL signal to the correct device. Probably you should look into your vendor, and set that up as well. First test it if you can wol from your router, after that test it cross vlan.

**Comment by ldlx** [comment]
WoL is a UDP packet sent to the broadcast address in your subnet/VLAN, so it is sent to all connected devices. As the WoL packet includes the MAC address of the device to be waked, every device decides if it should wake up or not.

You can send a UDP packet to a specific IP address in another subnet/VLAN, but the router of that subnet would not forward it to the broadcast address (when you don't set up an ARP proxy inside the router configuration like @topeminent).
You now need a computer in that subnet/VLAN that takes up your UDP packet and re-sends it to the broadcast address - a Wake-On-LAN proxy like this one: https://www.green-it-software.com/wake-on-lan-proxy-with-raspberry-pi/ (you have to compile it yourself on Raspberry Pi, x86, x64 - whatever you like).

**Comment by adgm1** [comment]
I am not sure to understand. What you suggest is to configure something in the firewall or in veyon.configurator?
Because what I realized is that if I use veyon master on a specific vlan I can power on any computer on the same vlan. For example I'm a teacher using ip 172.16.13.21 I can power on all the student's computers that have an ip address like 172.16.13.x. but the idea is to use veyon master on the 172.16.100.x and be able to power on any computer on vlan 10, 11, 12, ... (172.16.10.x, 172.16.11.x, 172.16.12.x...)

I need to add an exception for udp in firewall of each computer or just the one that try to access the others (172.16.100.x)?

**Comment by topeminent** [comment]
Your problem is at networking level, so the problem is not with veyon. The program sends out the WoL magic packet, but your switch simply drops it, because it s not in the same Vlan, and doesn t know what to do with the packet. Are you the person in charge of the network and firewall? Depends on your infrastructure, ( there s a possibility that there are multiple firewalls between the subnets) If you have static ip addresses , you can create a generic firewall rule to accept udp packets destination port 9 from ip address 172.16.13.100.0/24 to destination 172.16.0.0/16 . But your organizations firewall policy rules may be much more  constrained. It all depends on your environment.

**Comment by adgm1** [comment]
Yes Im in charge of the network. Our ip address are definied in DHCP for all the user's computers, only the Domain controler Server is in static IP. we also have an another firewall yes, but even if there is, I can not understand why the firewall is OK to send packet inside the same Vlan but not between different Vlan of the same network.
I tried to add UDP exception in windows firewall but I can not set ip adress, only port. So I don't know where should I configure the broadcast 172.16.10.255 or 172.16.11.255 in my server to allow magic packet and allow WOL in veyon
I can wake up machines from my server using the wolcmd command, so it seems the problem come from a configuration in VEYON. 
For example from 172.16.100.x I put this command "wolcmd B0-22-7A-XX-XX-XX 172.16.10.125 255.255.255.0 9" and the machine can wake up without probleme. So nothing is blocking the WOL from 172.16.100.x to 172.16.10.x

**Comment by adgm1** [comment]
I have exactly the same problem as this post [https://veyon.nodebb.com/topic/1201/wake-on-lan-across-vlans](url)
But I still not understand what should I do in veyon to make it work WOL in different VLAN

**Comment by topeminent** [comment]
When i referred to the firewall, i was referring to the firewall between the subnets(usually a router) The router should forwards the broadcasted packets to the correct vlan in order to wake up the target device. When you WOL in the same subnet, the packets won t even reach the firewall(router) because the switch will just forward the packets (depends on you switch model, it may be a lvl 3 switch, which is much more intelligent)

For why you can wol from 172.16.100.x to 172.16.10.125.x , much guess is because you have some forwarding set up correctly on that direction. You ll find you answer in your routers config.

**Comment by ldlx** [comment]
> For example from 172.16.100.x I put this command "wolcmd B0-22-7A-XX-XX-XX 172.16.10.125 255.255.255.0 9" and the machine can wake up without probleme.

Veyon does not "know" your different subnets. Veyon only sends to broadcast addresses of the ip subnets configured on your network devices.

**Comment by topeminent** [comment]
So nothing is blocking the WOL from 172.16.100.x to 172.16.10.x

**Comment by adgm1** [comment]
ok. So you are suggesting me to install a WOLproxy to do the job. What is the easiest one ? I never used it.
Or maybe I can change the ACLs configuration but this I can not do by myself

**Comment by adgm1** [comment]
> So nothing is blocking the WOL from 172.16.100.x to 172.16.10.x
> 
> You should draw a simple sketch about your network topology, What equipment(switch, router, firewall) is between vlan 100 and vlan 10.

I will ask this part. i think there is a swith for each vlan. the vlan 100 is only one switch for all servers. then the others vlans are distributed betwen other switchs

**Comment by topeminent** [comment]
i understand. But still, there should be an uplink from vlan 100 as well from vlan 10 to a point in the network topology where the two meet , and packets are forwarded between the two vlans. Remember the equipment model name as well.

**Comment by ldlx** [comment]
> ok. So you are suggesting me to install a WOLproxy to do the job. What is the easiest one ? I never used it.
I use this one: https://www.green-it-software.com/wake-on-lan-proxy-with-raspberry-pi/ because source code is available at no cost.
You just have to compile it, start the service and then you can forget about it. The steps are described in the readme.

**Comment by adgm1** [comment]
OK but this soft is for Linux. Is there one for windows? Thanks

**Comment by ldlx** [comment]
Maybe one of these:
https://www.mylanviewer.com/wake-on-lan-proxy-server.html
https://www.enviprot.com/de/produkt/wake-on-lan
https://codeberg.org/DecaTec/IoT.WakeOnLanProxy
